metastore/metastore-api/src/main/java/org/apache/drill/metastore/util/SchemaPathUtils.java

```diff
@@ -81,15 +106,50 @@ public static void addColumnMetadata(TupleMetadata schema, SchemaPath schemaPath
       names.add(colPath.getPath());
       colMetadata = schema.metadata(colPath.getPath());
       TypeProtos.MajorType pathType = types.get(SchemaPath.getCompoundPath(names.toArray(new String[0])));
+
+      boolean isDict = pathType != null && pathType.getMinorType() == TypeProtos.MinorType.DICT;
+      boolean isList = pathType != null && pathType.getMinorType() == TypeProtos.MinorType.LIST;
+      String name = colPath.getPath();
```

Comment:
More bad code smell. See notes above.

BTW: If you have to do a short-term fix, then this kind of error-prone, redundant code is fine. But, if we have time to fix the underlying problem, let's do it. Else, we can fix the problem in another PR.